### PR TITLE
[wasm][debugger] Tie sdb agent lifetime to the ExecutionContext and simplify the api

### PR DESCRIPTION
**Pull request by lewing**
Rework MonoSdbHelper and tie its lifetime to the ExecutionContext to simplify session handling.

### CODE REVIEW

**Comment by lewing** [comment]
@thaystg the smaller version is smaller but...

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs [hunk 1]
```diff
@@ -378,11 +378,23 @@ public TypeInfoWithDebugInformation(TypeInfo typeInfo, int debugId, string name)
...
             HasError = hasError;
         }
 
+        public MonoBinaryReader(byte [] data) : base(new MemoryStream(data)) {}
+
+        public static MonoBinaryReader From(Result result)
+        {
+            byte[] newBytes = Array.Empty<byte>();
+            if (!result.IsErr) {
+                newBytes = Convert.FromBase64String(result.Value?["result"]?["value"]?["value"]?.Value<string>());
```
Comment (radical) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:393]: This can NRE

Comment (lewing) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs:393]: The code has just moved, it is the same as before.

**Review by thaystg — APPROVED** [review]

**Comment by lewing** [comment]
Needs a little more work, the merge was slightly off.  Working on it now

**Comment by lewing** [comment]
should be good now.

**Pull request merged**